web/main/migrations/0035_auto_20200113_2106.py

```diff
@@ -0,0 +1,31 @@
+# Generated by Django 2.2.9 on 2020-01-13 21:06
+from django.db import migrations
+from ..utils import clone_model_instance
+
+def copy_common_resources(apps, schema_editor):
+    ContentNode = apps.get_model('main', 'ContentNode')
+    resource_types = set(['Link', 'TextBlock'])
+    for resource_type in resource_types:
+        resource_class = apps.get_model('main', resource_type)
+        seen = set()
+        for cn in ContentNode.objects.filter(resource_type=resource_type):
+            if cn.resource_id:
+                if cn.resource_id in seen:
+                    duplicate = clone_model_instance(resource_class.objects.filter(id=cn.resource_id).first())
+                    duplicate.save()
+                    cn.resource_id = duplicate.id
+                    cn.save()
+                seen.add(cn.resource_id)
+
+def cannot_reverse():
+    pass
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('main', '0034_auto_20200108_2017'),
+    ]
+
+    operations = [
+        migrations.RunPython(copy_common_resources, cannot_reverse)
```

Comment:
> Django makes a migrations.operations.RunPython.noop method available for irreversible migrations 

Just for nitpicky clarification, `migrations.RunPython.noop` isn't for irreversible migrations. It does work great for this usecase, where the intent is "there is no need to do any work when reversing this migration -- things will run fine if the database is rewound to migration 0034." If what you mean instead is "once this is done, it cannot be undone and you're never going back to migration 0034 again, get used to migration 0036 because that's where you live now," that's expressed by leaving the second parameter blank.

Reply:
Noop is good and straightforward.

Reply:
Thanks for correcting my sloppy language, @jcushman! Well-said.